frontend/src/App.res

```diff
@@ -18,19 +18,44 @@ query ($startDate: timestamp!, $endDate: timestamp!) {
   }
 `)
 
-let comparePulls = ((pn1, _b1), (pn2, _b2)) => {
-  -compare(pn1, pn2)
+module PullCompare = Belt.Id.MakeComparable({
+  type t = (int, option<string>)
+  let cmp = compare
+})
+
+let collectBenchmarksForRepo = (~repo, data: array<GetBenchmarks.t_benchmarks>): array<
+  GetBenchmarks.t_benchmarks,
+> => {
+  data->Belt.Array.keep(item => item.repo_id == repo)
 }
 
-let collectPulls = (data: array<GetBenchmarks.t_benchmarks>): array<(int, option<string>)> => {
-  let data = Belt.List.fromArray(data)
-  let data = Belt.List.keepMap(data, (item: GetBenchmarks.t_benchmarks) =>
+let collectPullsForRepo = (~repo, benchmarks: array<GetBenchmarks.t_benchmarks>): array<(
+  int,
+  option<string>,
+)> => {
+  benchmarks
+  ->collectBenchmarksForRepo(~repo)
+  ->Belt.Array.keepMap((item: GetBenchmarks.t_benchmarks) =>
     Belt.Option.flatMap(item.pull_number, pull_number => Some(pull_number, item.branch))
   )
-  let data = List.sort_uniq(comparePulls, data)
-  Belt.List.toArray(data)
+  ->Belt.Set.fromArray(~id=module(PullCompare))
+  ->Belt.Set.toArray
+}
+
+let collectRepos = (benchmarks: array<GetBenchmarks.t_benchmarks>): array<string> => {
```

Comment:
On a second thought, why not use GraphQL now? It seems to complicate quite a bit data filtering. Would it be possible to make another query to obtain the distinct list of repositories?

Reply:
I agree it would be better to query for specific data at different places instead of querying everything and filtering on the frontend side. I would prefer to address that in another PR though if that's ok for you?

Reply:
That's fine, of course.